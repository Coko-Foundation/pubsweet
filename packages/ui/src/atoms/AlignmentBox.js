import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const lightGrey = '#b3b3b3'
const mainGrey = '#666'
const whiteRGBA = 'rgba(255, 255, 255, 1)'

const borderRule = position => props => `
    border-${position}-width: ${props.noBorder[position] ? '0' : '1px'}
  `

const Root = styled.div.attrs({
  role: 'presentation',
})`
  box-shadow: inset 0 0 0 2px ${whiteRGBA};
  cursor: pointer;
  height: 26px;
  width: 17px;

  border-style: solid;
  border-color: ${mainGrey};
  ${borderRule('top')};
  ${borderRule('right')};
  ${borderRule('bottom')};
  ${borderRule('left')};

  background-color: ${props => (props.active ? mainGrey : 'transparent')};

  &:hover {
    background-color: ${lightGrey};
  }
`

const AlignmentBox = ({ active, id, noBorder, onClick }) => (
  <Root active={active} id={id} noBorder={noBorder} onClick={onClick} />
)

AlignmentBox.propTypes = {
  active: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  noBorder: PropTypes.shape({
    top: PropTypes.bool,
    bottom: PropTypes.bool,
    right: PropTypes.bool,
    left: PropTypes.bool,
  }),
  onClick: PropTypes.func,
}

AlignmentBox.defaultProps = {
  noBorder: {
    top: false,
    bottom: false,
    right: false,
    left: false,
  },
  onClick: () => null,
}

export default AlignmentBox
