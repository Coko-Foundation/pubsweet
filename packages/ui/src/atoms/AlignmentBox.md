Inactive

```js

const active = false;
const id = 'eyedee';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Active

```js

const active = true;
const id = 'eyedee';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Active with no right border

```js

const active = true;
const id = 'eyedee';
const noBorder = { right: true };
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```
