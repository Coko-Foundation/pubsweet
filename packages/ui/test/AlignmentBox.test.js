import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import AlignmentBox from '../src/atoms/AlignmentBox'

const myMock = jest.fn()
const props = {
  active: true,
  id: 'left',
  onClick: myMock,
  noBorder: {
    top: true,
    bottom: true,
    right: true,
    left: true,
  },
}

const requiredProps = {
  active: false,
  id: 'left',
}

const wrapper = shallow(<AlignmentBox {...requiredProps} />)
const wrapperMounted = mount(<AlignmentBox {...requiredProps} />)

describe('AlignmentBox', () => {
  test('is rendered correctly', () => {
    const tree = renderer.create(<AlignmentBox {...requiredProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('gets the correct props', () => {
    const noBorderDefaults = {
      top: false,
      bottom: false,
      right: false,
      left: false,
    }

    expect(wrapperMounted.prop('active')).toEqual(requiredProps.active)
    expect(wrapperMounted.prop('id')).toEqual(requiredProps.id)
    expect(wrapperMounted.prop('noBorder')).toEqual(noBorderDefaults)
  })

  test('with default props has borders and has not active indicator', () => {
    expect(wrapper.is('.active')).toBe(false)
    expect(wrapper.is('.noBorderTop')).toBe(false)
    expect(wrapper.is('.noBorderRight')).toBe(false)
    expect(wrapper.is('.noBorderBottom')).toBe(false)
    expect(wrapper.is('.noBorderLeft')).toBe(false)
  })

  test('with given props has the correct classes', () => {
    const newWrapper = shallow(<AlignmentBox {...props} />)

    expect(newWrapper.is('.active')).toBe(true)
    expect(newWrapper.is('.noBorderTop')).toBe(true)
    expect(newWrapper.is('.noBorderRight')).toBe(true)
    expect(newWrapper.is('.noBorderBottom')).toBe(true)
    expect(newWrapper.is('.noBorderLeft')).toBe(true)
  })

  test('on click the corresponding method is triggered', () => {
    const newWrapper = shallow(<AlignmentBox {...props} />)
    newWrapper.simulate('click')
    expect(newWrapper.instance().props.onClick).toHaveBeenCalled()
  })
})
