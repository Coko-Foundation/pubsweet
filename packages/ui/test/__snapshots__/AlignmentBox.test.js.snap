// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentBox is rendered correctly 1`] = `
.c0 {
  box-shadow: inset 0 0 0 2px rgba(255,255,255,1);
  cursor: pointer;
  height: 26px;
  width: 17px;
  border-style: solid;
  border-color: #666;
  border-top-width: 1px;
  border-right-width: 1px;
  border-bottom-width: 1px;
  border-left-width: 1px;
  background-color: transparent;
}

.c0:hover {
  background-color: #b3b3b3;
}

<div
  className="c0"
  id="left"
  onClick={[Function]}
  role="presentation"
/>
`;
